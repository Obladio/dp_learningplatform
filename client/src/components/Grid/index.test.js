import React from "react"
import ReactDOM from "react-dom"
import renderer from "react-test-renderer"
import {Container, Row, Col} from "./"

it("renders correctly", () => {
  const tree = renderer.create(<Container />)

  expect(tree).toMatchSnapshot()
})

it("renders correctly with params", () => {
  const tree = renderer.create(<Container className="randomClass" fluid />)

  expect(tree).toMatchSnapshot()
})

it("Row renders correctly", () => {
  const tree = renderer.create(<Row />)

  expect(tree).toMatchSnapshot()
})

it("Row renders correctly with params", () => {
  const tree = renderer.create(
    <Row
      className="randomClass"
      alignItems="center"
      justifyContent="end"
      noGutters
    />
  )

  expect(tree).toMatchSnapshot()
})

it("Col renders correctly", () => {
  const tree = renderer.create(<Col />)

  expect(tree).toMatchSnapshot()
})

it("Col renders correctly with params", () => {
  const tree = renderer.create(
    <Col
      className="randomClass"
      size={12}
      sm={12}
      md={12}
      lg={12}
      xl={12}
      offset={1}
      offsetSm={1}
      offsetMd={1}
      offsetLg={1}
      offsetXl={1}
      push={1}
      pushSm={1}
      pushMd={1}
      pushLg={1}
      pushXl={1}
      pull={1}
      pullSm={1}
      pullMd={1}
      pullLg={1}
      pullXl={1}
      alignSelf="end"
      flex="end"
      alignItems="center"
      justifyContent="center"
    />
  )

  expect(tree).toMatchSnapshot()
})

it("renders whole grid correctly", () => {
  const tree = renderer.create(
    <Container>
      <Row>
        <Col>
          <p>Test</p>
        </Col>
        <Col>
          <p>Test</p>
        </Col>
      </Row>
      <Row>
        <Col>
          <p>Test</p>
        </Col>
        <Col>
          <p>Test</p>
        </Col>
      </Row>
    </Container>
  )

  expect(tree).toMatchSnapshot()
})
