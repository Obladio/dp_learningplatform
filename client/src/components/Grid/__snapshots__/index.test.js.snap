// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col renders correctly 1`] = `
<div
  className="col-12"
/>
`;

exports[`Col renders correctly with params 1`] = `
<div
  className="randomClass col-12 col-sm-12 col-md-12 col-lg-12 col-xl-12 offset-1 offset-sm-1 offset-md-1 offset-lg-1 offset-xl-1 push-1 push-sm-1 push-md-1 push-lg-1 push-xl-1 pull-1 pull-sm-1 pull-md-1 pull-lg-1 pull-xl-1 align-self-end flex-end align-items-center justify-content-center"
/>
`;

exports[`Row renders correctly 1`] = `
<div
  className="row"
/>
`;

exports[`Row renders correctly with params 1`] = `
<div
  className="row randomClass align-items-center justify-content-end no-gutters"
/>
`;

exports[`renders correctly 1`] = `
<div
  className="container-default"
/>
`;

exports[`renders correctly with params 1`] = `
<div
  className="container-fluid randomClass"
/>
`;

exports[`renders whole grid correctly 1`] = `
<div
  className="container-default"
>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <p>
        Test
      </p>
    </div>
    <div
      className="col-12"
    >
      <p>
        Test
      </p>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <p>
        Test
      </p>
    </div>
    <div
      className="col-12"
    >
      <p>
        Test
      </p>
    </div>
  </div>
</div>
`;
